topdown/http.go

```diff
@@ -67,17 +68,51 @@ func builtinHTTPSend(bctx BuiltinContext, args []*ast.Term, iter func(*ast.Term)
 		return handleBuiltinErr(ast.HTTPSend.Name, bctx.Location, err)
 	}
 
+	usePersistentCache, err := getBoolValFromReqObj(req, "cache")
+	if err != nil {
+		return handleHTTPSendErr(bctx, err)
+	}
+
+	forceJSONDecode, err := getBoolValFromReqObj(req, "force_json_decode")
+	if err != nil {
+		return handleHTTPSendErr(bctx, err)
+	}
+
+	httpReq, client, err := createHTTPRequest(bctx, req)
+	if err != nil {
+		return handleHTTPSendErr(bctx, err)
+	}
+
 	// check if cache already has a response for this query
-	resp := checkHTTPSendCache(bctx, req)
+	var resp ast.Value
+
+	if usePersistentCache {
+		resp, err = checkHTTPSendPersistentCache(bctx, req, httpReq, client, forceJSONDecode)
+		if err != nil {
+			return handleHTTPSendErr(bctx, err)
+		}
+	} else {
+		resp = checkHTTPSendCache(bctx, req)
+	}
+
 	if resp == nil {
+		var size int
 		var err error
-		resp, err = executeHTTPRequest(bctx, req)
+
+		resp, size, err = executeHTTPRequest(httpReq, client, forceJSONDecode)
 		if err != nil {
 			return handleHTTPSendErr(bctx, err)
 		}
 
 		// add result to cache
```

Comment:
nit: similar to other comment. Delete this.

Reply:
Removed